### ISSUE TITLE
add ability to set custom opt out message when creating any new campaign

### ISSUE THREAD

**Issue by joemcl** [ProgressiveCoders/Spoke#632]
Currently the only way to edit/update the default opt out message is with a PR and commit to a fork of the core code at for example https://github.com/MoveOnOrg/Spoke/blob/aae35f862ae2097434afcd3225e0620df864aa33/src/containers/AssignmentTexterContact.jsx#L214 . 
A user organization may want to use it's own default opt out message, and a different one depending on the campaign. 
So let's add the ability to set a custom opt out message when creating any new campaign, or to use a generic, default opt out message for each campaign, for each user organization.
cc: @codygordon .

**Comment by deasterdaywfp** [comment]
@codygordon can we talk about adding this to the list of feature requests that are on your plate for the next week/two?

**Comment by schuyler1d** [comment]
This can be done at the instance level somewhat easily with a custom environment variable override, e.g. `process.env.DEFAULT_OPT_OUT_REPLY`

That would be a good-first-issue for someone.  A more complicated version would be to customize this per-organization or per-campaign, which would then load info from the organization.features column and have a place to edit/save it.

**Comment by ibrand** [comment]
@joemcl and @deasterdaywfp I'm cleaning up our `good first issues`s and want to know if this one was completed (since it's pretty old) or if it's open for new contributors to claim! If it's still open, do we have a design in mind for this?

**Comment by joemcl** [comment]
@ibrand it was not completed, it's still desired so yes open to claim. IMHO it would be something that can be added as part of a 'script' for each new Campaign, along with Interactions and Canned, especially since we now can easily load up/import a script from Google Docs. cc: @deasterdaywfp @lperson .

**Comment by ibrand** [comment]
Closing this because I see a way to edit the opt out message in the settings panel:
<img width="1440" alt="Screen Shot 2020-05-18 at 3 06 31 PM" src="https://user-images.githubusercontent.com/3953117/82250315-3aec0580-9919-11ea-8328-a491377f1176.png">

**Comment by joemcl** [comment]
@ibrand sorry if the Issue description is not clear. The idea is also to be able to have a *unique* opt-out message *per Campaign*  if desired. We can't do that, yet, so I'm reopening the Issue.

**Comment by ibrand** [comment]
Oh thanks @joemcl ! Appreciate that.

This would be a good task for a volunteer! It would take a little discussion / planning to decide where the campaign level opt out should go.

**Comment by joemcl** [comment]
Thanks @ibrand . I'm thinking a setting to be able to add a unique opt-out message, per Campaign, if desired, would be handy, via, for example, adding it to a Google Docs script import, or, to just use the org's default opt-out message if that is not included. Thoughts @lperson @deasterdaywfp @suegibson ?

**Comment by suegibson** [comment]
Custom opt out confirmations would be great!

**Comment by deasterdaywfp** [comment]
Agreed, this would be useful, but I do see this as a "nice to have" not a
need at the moment.

*Dylan Easterday*
*National Volunteer & Distributed Organizing Manager - Working Families
Party*
WFP4theMany.org
*Text *(347) 460-9451 or Schedule https://calendly.com/deasterday
Email deasterday@workingfamilies.org
Sign up to volunteer here <http://www.wfp4themany.org>

**Comment by arena** [comment]
The new texter UI lets you toggle between default and blank opt-out message... it could have a third or other options and a default per campaign
